Neos.ContentRepository/src/Infrastructure/Property/PropertyConverter.php

```diff
@@ -65,20 +79,43 @@ public function serializePropertyValues(
                     );
                 }
 
-                $serializedPropertyValues[$propertyName] = new SerializedPropertyValue(
+                return new SerializedPropertyValue(
                     $propertyValue,
                     (string)$propertyType
                 );
             } else {
-                if (array_key_exists($propertyName, $nodeType->getProperties())) {
-                    // $propertyValue == null and defined in node types -> we want to set the $propertyName to NULL
-                    $serializedPropertyValues[$propertyName] = new SerializedPropertyValue(null, (string)$propertyType);
-                } else {
-                    // $propertyValue == null and not defined in NodeTypes -> we want to unset $propertyName!
-                    $serializedPropertyValues[$propertyName] = null;
-                }
+                // $propertyValue == null and defined in node types (we have a resolved type)
+                // -> we want to set the $propertyName to NULL
+                return new SerializedPropertyValue(null, (string)$propertyType);
+            }
+        } catch (PropertyTypeIsInvalid $exception) {
+            if ($exception->getCode() === 1630063406 && $propertyValue === null) {
```

Comment:
can we somehow change this code? Especially checking for an exception code is weird here IMHO

Reply:
I actually resurrected that part of code from a dead path...
Maybe we can simply remove the outer try/catch. 
The original idea from $back then is that by setting a non existent property to null we would be able to unset the property altogether. That didn't survive the introduction of property checks though and has zero test coverage. So... should we just drop it? Unsetting obsolete properties is covered by integrity checks I think

Reply:
The "unset properties" feature has been restored on command level, tests will follow in a later PR

Reply:
@nezaniel awesome :) Thanks!